cmd/apprepository-controller/controller.go

```diff
@@ -367,26 +385,40 @@ func (c *Controller) handleObject(obj interface{}) {
 			return
 		}
 
+		if apprepo.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
This is required so that we don't re-process (ie. re-create cronjobs etc.) for an app repo that is being deleted.